### PR TITLE
Sds 213/form select component

### PR DESCRIPTION
**Pull request by akdetrick**
#### Related issues
Fixes https://meetup.atlassian.net/browse/SDS-213

#### Description
Adds `select` input component.

#### Screenshots (if applicable)
screen shot 2017-04-11 at 6 30 50 pm
screen shot 2017-04-11 at 6 30 59 pm
screen shot 2017-04-11 at 6 31 05 pm

### CODE REVIEW

File: src/forms/SelectInput.jsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+					{label}
+				</label>
+
+				<select
+					name={name}
+					required={required}
+					className={classNames}
+					onChange={this.onChange}
+					value={this.state.value}
+					{...other}>
```
Comment (mmcgahan) [inline src/forms/SelectInput.jsx]: putting the closing `>` at the same indentation level as the opening `<` will show that the `{` on the line below needs an extra indentation level.

File: src/forms/selectInput.test.jsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+		component = null;
+	});
+
+	it('exists', () => {
+		expect(() => TestUtils.findRenderedDOMComponentWithTag(component, 'select')).not.toThrow();
+	});
+
+	it('should have a NAME attribute', () => {
+		const selectEl = TestUtils.findRenderedDOMComponentWithTag(component, 'select');
+		expect(selectEl.name).toEqual('testSelect');
```
Comment (mmcgahan) [inline src/forms/selectInput.test.jsx]: in the future, set any 'expected' string values to constants to enforce a connection between the input and output

File: src/forms/selectInput.test.jsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+	});
+
+	it('should have default value in state', () => {
+		expect(!!component.state.value).toBe(true);
+	});
```
Comment (mmcgahan) [inline src/forms/selectInput.test.jsx]: I think the following test makes this test redundant

File: src/forms/selectInput.test.jsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+				name='testSelect'
+				options={testOptions}
+				onChange={onChange}
+			/>
+		);
+		const selectEl = TestUtils.findRenderedDOMComponentWithTag(component, 'select');
+
+		TestUtils.Simulate.change(selectEl, { target: { value: newValue } });
+
+		expect(changeSpy).toHaveBeenCalled();
```
Comment (mmcgahan) [inline src/forms/selectInput.test.jsx]: always good to confirm that the `changeSpy` has _not_ been called _before_ the simulated change event.

**Comment by akdetrick** [comment]
bump; orgX will need this soon

File: src/forms/SelectInput.jsx [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+				>
+					{
+						options.map((option, key) =>
+							<option key={key} value={option.value}>{option.label}</option>
+						)
+					}
+				</select>
+
+				{
+					errors &&
```
Comment (mmcgahan) [inline src/forms/SelectInput.jsx]: this might be safer as `errors && errors.length > 0`. not a blocker

**Comment by mmcgahan** [comment]
merge away

**Pull request merged**